### PR TITLE
✨ amp-ad type=dynad

### LINKED ISSUE #22968: I2I: amp-ad type=dynad
## Summary
Provides support for amp-ad type=dynad
[DynAd](https://www.dynad.net/) is an ad network used by https://afiliados.uol.com.br/

### PR DESCRIPTION
**Pull request by eduardogoncalves**
Provides support for amp-ad type=dynad
DynAd is an ad network used by https://afiliados.uol.com.br/
close #22968

### CODE REVIEW

**Comment by eduardogoncalves** [comment]
cc @jridgewell

File: ads/dynad.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<!---
+Copyright 2015 The AMP HTML Authors. All Rights Reserved.
```
Comment (torch2424) [inline ads/dynad.md]: Incorrect copyright year 😄

File: ads/_config.js [hunk 2]
```diff
@@ -396,7 +396,9 @@ export const adConfig = {
     prefetch: 'https://amp.ad.dotandad.com/dotandadsAmp.js',
     preconnect: 'https://bal.ad.dotandad.com',
   },
-
+  'dynad': {
+    preconnect: ['https://t.dynad.net', 'https://tm.jsuol.com.br'],
```
Comment (torch2424) [inline ads/_config.js]: So I noticed here you are not implementing `renderStartImplemented`. If possible, could you please implement this? It helps us serve ads as fast as possible to our users, and gives us some big performance gains.

Though, from your PR description, I'm not sure if you are making the PR on `dynad`'s behalf, can can implement that server side for them?

Let me know what you think. Thank you! 😄

**Comment by eduardogoncalves** [comment]
No, I don't work for DynAd nor at UOL, I'm externally submitting this to learn how to make an ad provider compatible with AMP.

File: ads/dynad.md (hunk 1 above)
Comment (eduardogoncalves) [reply ads/dynad.md]: fixed 😄

File: ads/dynad.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+import {validateSrcContains, validateSrcPrefix, writeScript} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function dynad(global, data) {
+  const adsrc = data.src;
```
Comment (calebcordry) [inline ads/dynad.js]: Can you use validateData here, it should eliminate the need for the if statement

https://github.com/ampproject/amphtml/blob/master/3p/3p.js#L225

Comment (eduardogoncalves) [reply ads/dynad.js]: @calebcordry does validateData validates src attribute?

Comment (calebcordry) [reply ads/dynad.js]: @eduardogoncalves yes it does.

**Comment by calebcordry** [comment]
@eduardogoncalves Looks like there are a few lint errors. You can try running `gulp lint --local_changes --fix` make sure the output looks good and commit. Let me know if there are any issues. Thanks!

**Pull request merged**